sbncode/CAFMaker/CAFMaker_module.cc

```diff
@@ -341,8 +340,33 @@ void CAFMaker::beginJob() {
 
 //......................................................................
 void CAFMaker::beginRun(art::Run& run) {
-  // fDetID = geom->DetId();
-  fDet = (Det_t)1;//(Det_t)fDetID;
+  // Heuristic method to determine the detector ID
+  const geo::GeometryCore* geom = lar::providerFrom<geo::Geometry>();
+
+  std::string gdml = geom->GDMLFile();
+  gdml = basename(gdml.c_str()); // strip directory part
+  std::cout << "CAFMaker: Attempting to deduce detector from GDML file name: '" << gdml
+            << "' and configured detector name: '" << geom->DetectorName() << "'. ";
+  // Do we find the string in either of the names?
+  const bool hasSBND = ((gdml.find("sbnd") != std::string::npos) ||
+                        (geom->DetectorName().find("sbnd") != std::string::npos));
+  const bool hasIcarus = ((gdml.find("icarus") != std::string::npos) ||
+                          (geom->DetectorName().find("icarus") != std::string::npos));
```

Comment:
```
  const bool hasSBND = ((gdml.find("sbnd") != std::string::npos) ||
                        (geom->DetectorName().find("sbnd") != std::string::npos));
  const bool hasIcarus = ((gdml.find("icarus") != std::string::npos) ||
                          (geom->DetectorName().find("icarus") != std::string::npos));
```
While `geo:GeometryCore::DetectorName()` is guaranteed to be lower case, the GDML file name may have any case.
I _suggest_ the `gdml` string to be converted to lower case.